metricbeat/module/system/process/process.go

```diff
@@ -131,14 +131,17 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 		return err
 	} else {
 		proc, root, err := m.stats.GetOneRootEvent(m.setpid)
-		if err != nil {
+		if err != nil && !errors.Is(err, process.NonFatalErr{}) {
+			// return only if the error is fatal in nature
 			return fmt.Errorf("error fetching pid %d: %w", m.setpid, err)
+		} else if (err != nil && errors.Is(err, process.NonFatalErr{})) {
+			err = mb.PartialMetricsError{Err: err}
 		}
+		// if error is non-fatal, emit partial metrics.
 		r.Event(mb.Event{
 			MetricSetFields: proc,
 			RootFields:      root,
 		})
+		return err
 	}
-
-	return nil
```

Comment:
[Blocker]
Just to be sure I understood it right.
It does not return `nil` here anymore because now it'll return on the `if m.setpid == 0` section the `mb.PartialMetricsError`. And the same for the else branch, right?

Reply:
correct.